ui/src/app/workflows/components/workflows-toolbar/workflows-toolbar.tsx

```diff
@@ -0,0 +1,127 @@
+import {NotificationType} from 'argo-ui';
+import * as PropTypes from 'prop-types';
+import * as React from 'react';
+import {Workflow} from '../../../../models';
+import {AppContext, Consumer} from '../../../shared/context';
+import * as Actions from '../../../shared/workflow-actions';
+
+require('./workflows-toolbar.scss');
+
+interface WorkflowsToolbarProps {
+    selectedWorkflows: {[index: string]: Workflow};
+    loadWorkflows: () => void;
+    isDisabled: Actions.ActionDisabled;
+}
+
+interface WorkflowGroupAction {
+    action: () => void;
+    title: string;
+    disabled: boolean;
+    iconClassName: string;
+    className: string;
+}
+
+export class WorkflowsToolbar extends React.Component<WorkflowsToolbarProps, {}> {
+    public static contextTypes = {
+        router: PropTypes.object,
+        apis: PropTypes.object
+    };
+
+    constructor(props: WorkflowsToolbarProps) {
+        super(props);
+    }
+
+    public render() {
+        return (
+            <Consumer>
+                {ctx => (
+                    <div className='workflows-toolbar'>
+                        <div className='workflows-toolbar__count'>
+                            {this.noneSelected() ? 'No' : this.getNumberSelected()}
+                            &nbsp;workflow{this.getNumberSelected() === 1 ? '' : 's'} selected
+                        </div>
+                        <div className='workflows-toolbar__actions'>{this.renderActions(ctx)}</div>
+                    </div>
+                )}
+            </Consumer>
+        );
+    }
+
+    private noneSelected(): boolean {
+        return Object.keys(this.props.selectedWorkflows).length < 1;
+    }
+
+    private getNumberSelected(): number {
+        return Object.keys(this.props.selectedWorkflows).length;
```

Comment:
I could be wrong, but doesn't
```suggestion
        return this.props.selectedWorkflows.length;
```
work?

Reply:
I tried it, it doesn't - since `selectedWorkflows` is an object rather than an array it doesn't have a `length` property

Reply:
Great